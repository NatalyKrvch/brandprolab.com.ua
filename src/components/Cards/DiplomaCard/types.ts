export interface DiplomaCardProps {
  diplomaURL: string;
  onClick?: () => void;
}
