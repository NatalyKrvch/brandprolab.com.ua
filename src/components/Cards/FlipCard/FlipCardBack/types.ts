export interface FlipCardBackProps {
  header: string;
  list: string[];
}
