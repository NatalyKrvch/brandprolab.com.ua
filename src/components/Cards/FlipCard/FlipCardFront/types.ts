export interface FlipCardFrontProps {
  iconURL: string;
  header: string;
  label: string;
  description: string;
  callToAction: string;
}
