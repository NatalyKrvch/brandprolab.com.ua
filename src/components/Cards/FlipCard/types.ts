import type { ReactNode } from 'react';

export interface FlipCardProps {
  front: ReactNode;
  back: ReactNode;
}
