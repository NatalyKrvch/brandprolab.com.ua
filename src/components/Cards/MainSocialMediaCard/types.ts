export interface MainSocialMediaCardProps {
  desktopPhotoURL: string;
  tabletPhotoURL: string;
  mobilePhotoURL: string;
  iconURL: string;
  platformName: string;
  description: string;
  platformURL: string;
}
