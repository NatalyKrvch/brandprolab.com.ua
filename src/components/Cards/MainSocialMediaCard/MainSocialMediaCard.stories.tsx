import type { Meta, StoryObj } from '@storybook/react';

import MainSocialMediaCard from './MainSocialMediaCard';
import { MainSocialMediaCardProps } from './types';

const meta: Meta<typeof MainSocialMediaCard> = {
  title: 'Components/MainSocialMediaCard',
  component: MainSocialMediaCard,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    desktopPhotoURL: {
      control: 'text',
      description: 'URL of the desktop version of the photo.',
      table: { type: { summary: 'string' } },
    },
    tabletPhotoURL: {
      control: 'text',
      description: 'URL of the tablet version of the photo.',
      table: { type: { summary: 'string' } },
    },
    mobilePhotoURL: {
      control: 'text',
      description: 'URL of the mobile version of the photo (fallback).',
      table: { type: { summary: 'string' } },
    },
    iconURL: {
      control: 'text',
      description: 'URL of the social media icon.',
      table: { type: { summary: 'string' } },
    },
    platformName: {
      control: 'text',
      description: 'Name of the social media platform.',
      table: { type: { summary: 'string' } },
    },
    description: {
      control: 'text',
      description: 'Short description text under the title.',
      table: { type: { summary: 'string' } },
    },
    platformURL: {
      control: 'text',
      description: 'Link URL to the platform page.',
      table: { type: { summary: 'string' } },
    },
  },
};

export default meta;

type Story = StoryObj<MainSocialMediaCardProps>;

export const Default: Story = {
  args: {
    desktopPhotoURL: '/images/temp/photo_desktop.png',
    tabletPhotoURL: '/images/temp/photo_tablet.png',
    mobilePhotoURL: '/images/temp/photo_mobile.png',
    iconURL: '/icons/cards/socialMedia/insta.svg',
    platformName: 'Facebook',
    description: 'Connect with friends and the world around you.',
    platformURL: 'https://www.facebook.com',
  },
};
