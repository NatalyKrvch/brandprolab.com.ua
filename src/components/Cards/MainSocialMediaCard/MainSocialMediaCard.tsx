import { AdaptiveImage } from '@/components/AdaptiveImage';
import { PictureSource } from '@/components/AdaptiveImage/types';
import { Icon } from '@/components/Icon';
import { ExternalLink } from '@/components/Links';
import { IconClassType } from '@/lib/constants';
import {
  DESKTOP_MEDIA_QUERY,
  MainSocialCardColor,
  TABLET_MEDIA_QUERY,
} from '@/styles/constants';

import { SimpleCard } from '../SimpleCard';
import { MainSocialMediaCardProps } from './types';

const MainSocialMediaCard = ({
  desktopPhotoURL,
  tabletPhotoURL,
  mobilePhotoURL,
  iconURL,
  platformName,
  description,
  platformURL,
}: MainSocialMediaCardProps) => {
  const sources: PictureSource[] = [
    { srcSet: desktopPhotoURL, media: DESKTOP_MEDIA_QUERY },
    { srcSet: tabletPhotoURL, media: TABLET_MEDIA_QUERY },
  ];

  return (
    <ExternalLink href={platformURL}>
      <SimpleCard
        backgroundColor="bg-gradient-to-bl overflow-hidden tablet:bg-gradient-to-br desktop:bg-gradient-to-br from-teal-light to-gray-light"
        className="flex h-auto flex-wrap content-start items-start justify-between tablet:flex-row-reverse tablet:flex-nowrap tablet:items-center tablet:justify-center tablet:pr-8 desktop:relative desktop:flex-row-reverse desktop:flex-nowrap desktop:items-start desktop:justify-end desktop:gap-0 desktop:pr-8"
      >
        <div className="flex items-start gap-4 pl-4 pt-6 mobile:min-w-0 mobile:flex-1 desktop:absolute desktop:left-240 desktop:flex-col desktop:gap-6 desktop:pb-9 desktop:pr-9 desktop:pt-9">
          <Icon
            circleColor={`bg-${MainSocialCardColor.BLUE}`}
            iconURL={iconURL}
            type={IconClassType.SOCIAL_MEDIA_MAIN}
          />

          <div className="z-10">
            <p className="mb-1 text-28 font-bold leading-none tracking-normal text-blue-default tablet:text-4xl tablet:leading-none tablet:tracking-normal desktop:px-2 desktop:text-5xl desktop:leading-none desktop:tracking-normal">
              {platformName}
            </p>

            <p className="min-w-196 text-sm font-normal leading-4 tracking-normal text-gray-dark mobile:max-w-250 tablet:min-w-150 tablet:max-w-320 tablet:text-base tablet:leading-5 desktop:px-2 desktop:text-lg desktop:leading-22">
              {description}
            </p>
          </div>
        </div>

        <div className="ml-auto h-230 min-w-150 flex-shrink-0 overflow-hidden tablet:ml-0 tablet:h-220 desktop:ml-0 desktop:h-416">
          <AdaptiveImage
            fallbackSrc={mobilePhotoURL}
            sources={sources}
            alt="Photo"
            className="h-full w-auto object-contain"
          />
        </div>
      </SimpleCard>
    </ExternalLink>
  );
};

export default MainSocialMediaCard;
