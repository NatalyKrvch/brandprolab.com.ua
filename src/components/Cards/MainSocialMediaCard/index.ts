export { default as MainSocialMediaCard } from './MainSocialMediaCard';
