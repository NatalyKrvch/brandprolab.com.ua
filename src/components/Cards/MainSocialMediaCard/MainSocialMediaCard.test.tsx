import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';
import React from 'react';

import MainSocialMediaCard from './MainSocialMediaCard';
import { MainSocialMediaCardProps } from './types';

jest.mock('@/components/AdaptiveImage', () => ({
  __esModule: true,
  AdaptiveImage: ({
    fallbackSrc,
    alt,
  }: {
    fallbackSrc: string;
    alt: string;
  }) => <img src={fallbackSrc} alt={alt} />,
}));

jest.mock('@/components/Icon', () => ({
  __esModule: true,
  Icon: ({ circleColor, iconURL, type }: any) => (
    <svg
      role="img"
      aria-label="icon"
      data-circle-color={circleColor}
      data-icon-url={iconURL}
      data-type={type}
    />
  ),
}));

jest.mock('@/components/Links', () => ({
  __esModule: true,
  ExternalLink: ({
    href,
    children,
  }: {
    href: string;
    children: React.ReactNode;
  }) => <a href={href}>{children}</a>,
}));

jest.mock('../SimpleCard', () => ({
  __esModule: true,
  SimpleCard: ({ children }: { children: React.ReactNode }) => (
    <div>{children}</div>
  ),
}));

describe('MainSocialMediaCard', () => {
  const defaultProps: MainSocialMediaCardProps = {
    desktopPhotoURL: '/desktop.jpg',
    tabletPhotoURL: '/tablet.jpg',
    mobilePhotoURL: '/mobile.jpg',
    iconURL: '/icon.svg',
    platformName: 'Test Platform',
    description: 'Test description',
    platformURL: 'https://example.com',
  };

  it('renders a link with correct href and matches snapshot', () => {
    const { asFragment } = render(<MainSocialMediaCard {...defaultProps} />);
    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('href', defaultProps.platformURL);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the icon SVG with correct attributes and matches snapshot', () => {
    const { asFragment, container } = render(
      <MainSocialMediaCard {...defaultProps} />,
    );
    const icon = container.querySelector('svg[aria-label="icon"]');
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveAttribute('data-icon-url', defaultProps.iconURL);
    expect(icon).toHaveAttribute('data-circle-color');
    expect(icon).toHaveAttribute('data-type');
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays platformName and description text and matches snapshot', () => {
    const { asFragment } = render(<MainSocialMediaCard {...defaultProps} />);
    expect(screen.getByText(defaultProps.platformName)).toBeInTheDocument();
    expect(screen.getByText(defaultProps.description)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the fallback image with correct src and alt and matches snapshot', () => {
    const { asFragment } = render(<MainSocialMediaCard {...defaultProps} />);
    const img = screen.getByAltText('Photo') as HTMLImageElement;
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', defaultProps.mobilePhotoURL);
    expect(img).toHaveAttribute('alt', 'Photo');
    expect(asFragment()).toMatchSnapshot();
  });
});
