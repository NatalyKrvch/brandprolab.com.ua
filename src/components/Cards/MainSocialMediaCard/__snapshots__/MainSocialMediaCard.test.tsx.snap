// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSocialMediaCard displays platformName and description text and matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="https://example.com"
  >
    <div>
      <div
        class="flex items-start gap-4 pl-4 tablet:pl-0 pt-6 mobile:min-w-0 mobile:flex-1 desktop:absolute desktop:left-240 desktop:flex-col desktop:gap-6 desktop:pb-9 desktop:pr-9 desktop:pt-9"
      >
        <svg
          aria-label="icon"
          data-circle-color="bg-blue-default"
          data-icon-url="/icon.svg"
          data-type="social-media-main"
          role="img"
        />
        <div
          class="z-10"
        >
          <p
            class="mb-1 text-28 font-bold leading-none tracking-normal text-blue-default tablet:text-4xl tablet:leading-none tablet:tracking-normal desktop:px-2 desktop:text-5xl desktop:leading-none desktop:tracking-normal"
          >
            Test Platform
          </p>
          <p
            class="min-w-196 text-sm font-normal leading-4 tracking-normal text-gray-dark mobile:max-w-250 tablet:min-w-150 tablet:max-w-320 tablet:text-base tablet:leading-5 desktop:px-2 desktop:text-lg desktop:leading-22"
          >
            Test description
          </p>
        </div>
      </div>
      <div
        class="ml-auto h-230 min-w-150 flex-shrink-0 overflow-hidden tablet:ml-0 tablet:h-220 desktop:ml-0 desktop:h-416"
      >
        <img
          alt="Photo"
          src="/mobile.jpg"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`MainSocialMediaCard renders a link with correct href and matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="https://example.com"
  >
    <div>
      <div
        class="flex items-start gap-4 pl-4 tablet:pl-0 pt-6 mobile:min-w-0 mobile:flex-1 desktop:absolute desktop:left-240 desktop:flex-col desktop:gap-6 desktop:pb-9 desktop:pr-9 desktop:pt-9"
      >
        <svg
          aria-label="icon"
          data-circle-color="bg-blue-default"
          data-icon-url="/icon.svg"
          data-type="social-media-main"
          role="img"
        />
        <div
          class="z-10"
        >
          <p
            class="mb-1 text-28 font-bold leading-none tracking-normal text-blue-default tablet:text-4xl tablet:leading-none tablet:tracking-normal desktop:px-2 desktop:text-5xl desktop:leading-none desktop:tracking-normal"
          >
            Test Platform
          </p>
          <p
            class="min-w-196 text-sm font-normal leading-4 tracking-normal text-gray-dark mobile:max-w-250 tablet:min-w-150 tablet:max-w-320 tablet:text-base tablet:leading-5 desktop:px-2 desktop:text-lg desktop:leading-22"
          >
            Test description
          </p>
        </div>
      </div>
      <div
        class="ml-auto h-230 min-w-150 flex-shrink-0 overflow-hidden tablet:ml-0 tablet:h-220 desktop:ml-0 desktop:h-416"
      >
        <img
          alt="Photo"
          src="/mobile.jpg"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`MainSocialMediaCard renders the fallback image with correct src and alt and matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="https://example.com"
  >
    <div>
      <div
        class="flex items-start gap-4 pl-4 tablet:pl-0 pt-6 mobile:min-w-0 mobile:flex-1 desktop:absolute desktop:left-240 desktop:flex-col desktop:gap-6 desktop:pb-9 desktop:pr-9 desktop:pt-9"
      >
        <svg
          aria-label="icon"
          data-circle-color="bg-blue-default"
          data-icon-url="/icon.svg"
          data-type="social-media-main"
          role="img"
        />
        <div
          class="z-10"
        >
          <p
            class="mb-1 text-28 font-bold leading-none tracking-normal text-blue-default tablet:text-4xl tablet:leading-none tablet:tracking-normal desktop:px-2 desktop:text-5xl desktop:leading-none desktop:tracking-normal"
          >
            Test Platform
          </p>
          <p
            class="min-w-196 text-sm font-normal leading-4 tracking-normal text-gray-dark mobile:max-w-250 tablet:min-w-150 tablet:max-w-320 tablet:text-base tablet:leading-5 desktop:px-2 desktop:text-lg desktop:leading-22"
          >
            Test description
          </p>
        </div>
      </div>
      <div
        class="ml-auto h-230 min-w-150 flex-shrink-0 overflow-hidden tablet:ml-0 tablet:h-220 desktop:ml-0 desktop:h-416"
      >
        <img
          alt="Photo"
          src="/mobile.jpg"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`MainSocialMediaCard renders the icon SVG with correct attributes and matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="https://example.com"
  >
    <div>
      <div
        class="flex items-start gap-4 pl-4 tablet:pl-0 pt-6 mobile:min-w-0 mobile:flex-1 desktop:absolute desktop:left-240 desktop:flex-col desktop:gap-6 desktop:pb-9 desktop:pr-9 desktop:pt-9"
      >
        <svg
          aria-label="icon"
          data-circle-color="bg-blue-default"
          data-icon-url="/icon.svg"
          data-type="social-media-main"
          role="img"
        />
        <div
          class="z-10"
        >
          <p
            class="mb-1 text-28 font-bold leading-none tracking-normal text-blue-default tablet:text-4xl tablet:leading-none tablet:tracking-normal desktop:px-2 desktop:text-5xl desktop:leading-none desktop:tracking-normal"
          >
            Test Platform
          </p>
          <p
            class="min-w-196 text-sm font-normal leading-4 tracking-normal text-gray-dark mobile:max-w-250 tablet:min-w-150 tablet:max-w-320 tablet:text-base tablet:leading-5 desktop:px-2 desktop:text-lg desktop:leading-22"
          >
            Test description
          </p>
        </div>
      </div>
      <div
        class="ml-auto h-230 min-w-150 flex-shrink-0 overflow-hidden tablet:ml-0 tablet:h-220 desktop:ml-0 desktop:h-416"
      >
        <img
          alt="Photo"
          src="/mobile.jpg"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;
